packages/babel-plugin-transform-modules-amd/src/index.js

```diff
@@ -68,9 +69,17 @@ export default function(api, options) {
               }
             }
 
+            if (!loose) {
+              headers.push(...buildReexportsFromMeta(meta, metadata, loose));
+            }
+
             headers.push(
               ...buildNamespaceInitStatements(meta, metadata, loose),
             );
+
+            if (loose) {
+              headers.push(...buildReexportsFromMeta(meta, metadata, loose));
```

Comment:
I've lost a bit of context on all this, but couldn't we include this in `buildNamespaceInitStatements` when in loose mode, and leave the old logic where it was? I might also just be forgetting stuff, so either way no worries :P